cmos_noise_map/utils/read_write_utils.py

```diff
@@ -1,124 +1,138 @@
-        )
-        df.to_csv(filename, index=False)
-    else:
-        pass
-
-
-def read_parameter_table(csv_file):
-    """
-    A utility function to read in a parameter map
-
-    Parameters
-    ----------
-    csv_file : .csv file
-        The csv file containing the parameter data output.
-
-    Returns
-    -------
-    means : TYPE: list(float), length = num_peaks
-        DESCRIPTION: The means of each of the Gaussian modes calculated by GMM
-    var : TYPE: list(float), length = num_peaks
-        DESCRIPTION: The covariance of each Gaussian mode calculated by GMM
-    num_peaks : TYPE: int
-        DESCRIPTION: The number of Gaussians used to model the distribution of values of the pixel
-    amps : TYPE: list(float)
-        DESCRIPTION: The weights of each gaussian in the mixture. All weights sum to 1.
-
-    """
-    gmm_df = pd.read_csv(csv_file, header=0)
-    means = gmm_df["Means"]
-    var = gmm_df["Covariances"]
-    num_peaks = []
-    for i in gmm_df["Number of Peaks"]:
-        if not np.isnan(i):
-            num_peaks.append(int(i))
-        else:
-            num_peaks.append(np.nan)
-    amps = gmm_df["Mixture Weights"]
-    return means, var, num_peaks, amps
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Jan 24 13:42:45 2023
+
+@author: Prerana Kottapalli
+"""
+from astropy.io import fits
+from glob import glob
+import numpy as np
+import pandas as pd
+import os
+import tempfile
+
+from cmos_noise_map.utils.data_utils import qc_input
+
+def read_bias_frames(path: str, data_ext=0):
+    """
+    A function to open up fits files with memory mapping.
+
+    Parameters
+    ----------
+    path : str
+        DESCRIPTION. The path to your fits files without the .fits at the end
+    data_ext : integer or string, optional
+        DESCRIPTION. The default is 0. This is the extension to the data header data unit (hdu) that is to be processed.
+
+    Returns
+    -------
+    ims : array
+        DESCRIPTION. And array of opened, memory mapped, fits hdus. Not the data itself.
+
+    """
+    # Assumes trimmed and processed bias frames
+    files = glob(path + str("*.fits"), recursive=True)
+    if len(files)==0:
+        images = []
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            file_names = glob(path + str("*.fz"), recursive=True)
+            for filename in file_names:
+                base_filename, file_extension = os.path.splitext(os.path.basename(filename))
+                print(tmpdirname)
+                output_filename = os.path.join(tmpdirname, base_filename)
+                print(output_filename)
+                os.system('funpack -O {0} {1}'.format(output_filename, filename))
+                images.append(fits.open(output_filename, memmap=True, do_not_scale_image_data=True))
+    else:
+        images = [
+            fits.open(f, memmap=True, do_not_scale_image_data=True) for f in files
+        ]  # Doesn't work unless not scaled
+    qc_input(images)
```

Comment:
Nitpick: Given that this function is performing an action, i.e. checking the structure/number of input files provided, can we give it a more descriptive, action-oriented name? e.g. `check_input_data` or something to that effect